src/map/zone.h

```diff
@@ -678,7 +686,7 @@ class CZone
     void LoadZoneLines();
     void LoadZoneWeather();
 
-    CTreasurePool* m_TreasurePool;
+    std::unique_ptr<CTreasurePoolContainer> m_TreasurePoolContainer;
```

Comment:
If a treasure pool is just a wrapper around 2x unordered_maps and it is simple to initialize itself - there isn't a need for it to be a unique_ptr, it can just be a regular member of this class